hydrolib/core/dflowfm/tim/serializer.py

```diff
@@ -0,0 +1,120 @@
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+from hydrolib.core.basemodel import ModelSaveSettings, SerializerConfig
+
+TimeSeriesRow = List[str]
+TimeSeriesBlock = List[TimeSeriesRow]
+
+
+class TimSerializerConfig(SerializerConfig):
+    """Configuration settings for the TimSerializer."""
+
+    column_spacing: int = 1
+    """(int): The number of spaces to include between columns in the serialized .tim file."""
+
+
+class TimSerializer:
+    @staticmethod
+    def serialize(
+        path: Path,
+        data: Dict[str, Any],
+        config: TimSerializerConfig,
+        save_settings: ModelSaveSettings,
+    ) -> None:
+        """
+        Serializes timeseries data to a file in .tim format.
+
+        Args:
+            path (Path): The path to the destination .tim file.
+            data (Dict[str, Any]): The timeseries data to be serialized. The data should be provided as a dictionary with the following keys:
+                - 'comments' (List[str]): A list of comments to be included at the beginning of the file. Each comment should be a string.
+                - 'timeseries' (Dict[float, List[float]]): A dictionary representing the timeseries data, where the key is the time and the value the values at that time.
+            config (TimSerializerConfig): The serialization configuration settings.
+            save_settings (ModelSaveSettings): The save settings to be used.
+        """
+        path.parent.mkdir(parents=True, exist_ok=True)
+
+        commentlines = TimSerializer._serialize_comment_lines(data)
+        timeserieslines = TimSerializer._serialize_timeseries_lines(data, config)
+
+        file_content = TimSerializer._serialize_file_content(
+            timeserieslines, commentlines
+        )
+        with path.open("w") as file:
+            file.write(file_content)
+
+    @staticmethod
+    def _serialize_comment_lines(data) -> List[str]:
+        commentlines = []
+        for comment in data["comments"]:
+            commentlines.append(f"#{comment}")
+        return commentlines
+
+    @staticmethod
+    def _serialize_timeseries_lines(data, config) -> List[str]:
+        format_float = lambda v: f"{v:{config.float_format}}"
```

Comment:
Why not define this lambda in the `_serialize_to_timeseries_block` method?

Reply:
since _serialize_to_timeseries_block only needs format_float, passing config in that method would be passing too much information into the method.

Reply:
But then you could also just pass the `config.float_format` to the method and define the lambda inside the method. 
Passing functions normally is a bit of a code smell.
But since it is all internal/private, lets keep it like this.